docs/safe-haven-services/s3-service.md

```diff
@@ -0,0 +1,214 @@
+os.environ["http_proxy"] = ""
+resource = boto3.resource("s3")
+bucket = resource.Bucket("dummydata")
+bucket.download_file("326834963428524640226726425259803542053/249177910747091225438117569123869339900/MR.304084489533501143843524990882920225135-an.dcm", "downloaded.dcm")
+```
+
+### Load object into pydicom dataset
+
+```console
+pip install pydicom
+```
+
+This will load the data into a pydicom dataset without actually storing a file on disk:
+```py
+import boto3
+import io
+import os
+import pydicom
+
+os.environ["http_proxy"] = ""
+
+resource = boto3.resource("s3")
+bucket = resource.Bucket("dummydata")
+obj = bucket.Object("326834963428524640226726425259803542053/249177910747091225438117569123869339900/MR.304084489533501143843524990882920225135-an.dcm")
+dcm_bytes = io.BytesIO(obj.get()["Body"].read())
+ds = pydicom.dcmread(dcm_bytes)
+
+print(ds["StudyInstanceUID"])
+# (0020,000D) Study Instance UID                  UI: 1.2.840.113619.2.411.3.4077533701.216.1476084945.95
+```
+
+## R Usage
+
+There are three different packages for R, you can install `s3` or `aws.s3` or `paws`, but `aws.s3` is recommended and documented here.
+
+You should have the environment variables set, either in your terminal (see above), or in your `.Renviron` file, or explicitly in your R script.
+
+Note! `aws.s3` uses a different name `AWS_S3_ENDPOINT`.
+
+To set environment variables in your `.Renviron` file:
+```
+AWS_ACCESS_KEY_ID=<my access key id>
+AWS_SECRET_ACCESS_KEY=<my secret key>
+AWS_ENDPOINT_URL="http://nsh-fs02:7070"
+AWS_S3_ENDPOINT="http://nsh-fs02:7070"
+AWS_DEFAULT_REGION="us-east-1"
+```
+
+To set environment variables in your script:
+```
+Sys.setenv( AWS_ENDPOINT_URL="http://nsh-fs02:7070" )
+Sys.setenv( AWS_S3_ENDPOINT="http://nsh-fs02:7070" )
+Sys.setenv( AWS_DEFAULT_REGION="us-east-1" )
+```
+
+Note! Only *after* you have installed packages with `install.packages()` you should use
+```
+Sys.setenv( http_proxy="" )
+```
+
+An example R script to download a file from S3:
+```
+library(aws.s3)
+my_bucket <- "dummydata"
+my_access_key <- "dummydata"
+my_secret_key <- "abigsecretkey"
+my_region <- "us-east-1"
+my_endpoint <- "http://nsh-fs02:7070"
+my_endpoint_host <- "nsh-fs02:7070"
+my_object_path <- "326834963428524640226726425259803542053/249177910747091225438117569123869339900/MR.304084489533501143843524990882920225135-an.dcm"
+Sys.setenv( AWS_ENDPOINT_URL="http://nsh-fs02:7070" )
+Sys.setenv( AWS_S3_ENDPOINT="http://nsh-fs02:7070" )
+Sys.setenv( AWS_DEFAULT_REGION="us-east-1" )
+Sys.setenv( http_proxy="" )
+save_object( my_object_path, file = "downloaded.dcm", bucket = my_bucket, base_url=my_endpoint_host, region="", use_https=FALSE, key = my_access_key, secret = my_secret_key )
+```
+
+Note! You need to have the region set in the environment variable and pass `region=""` to the functions, otherwise you get a `cannot resolve host` error. The `base_url` is just the host and port, no `http://` prefix, and `use_https` is false.
```

Comment:
Note formatting is inconsistent within this page as well as compared with the rest of the docs. For important pieces of text, [this](https://docs.eidf.ac.uk/services/gpuservice/policies/#kubernetes-job-names) is probably the most visible.